### PR TITLE
docs(vrl): Documentation for redact redactor option

### PR DESCRIPTION
**Pull request by tmccombs**

### CODE REVIEW

File: website/cue/reference/remap/functions/redact.cue [hunk 1]
```diff
@@ -56,6 +56,39 @@ remap: functions: redact: {
...
+			name: "redactor"
+			description: #"""
+				Specifies what to replace the redacted strings with.
+
+				It is given as an object with a "type" key specifying the type of redactor to use
+				and additional keys depending on the type. The following types are supported:
+
+				- `full`: The default. Replace with the string "[REDACTED]".
+				- `text`: Replace with a custom string. The `replacement` key is required, and must
+				  contain the string that will be used as a replacement.
```
Comment (maycmlee) [inline website/cue/reference/remap/functions/redact.cue:69]:
```suggestion
				  contain the string that is used as a replacement.
```

File: website/cue/reference/remap/functions/redact.cue [hunk 2]
```diff
@@ -56,6 +56,39 @@ remap: functions: redact: {
...
+				- `sha2`: Hash the redacted text with SHA-2 as with [`sha2`](\(urls.sha2)). Supports two optional parameters:
+					- `variant`: The variant of the algorithm to use. Defaults to "SHA-512/256".
+					- `encoding`: How to encode the hash as text. Can be "base16" or "base64".
+						Defaults to "base64".
+				- `sha3`: Hash the redacted text with SHA-3 as with [`sha3`](\(urls.sha3)). Supports two optional parameters:
+					- `variant`: The variant of the algorithm to use. Defaults to "SHA3-512".
+					- `encoding`: How to encode the hash as text. Can be "base16" or "base64".
+						Defaults to "base64".
```
Comment (maycmlee) [inline website/cue/reference/remap/functions/redact.cue:77]:
```suggestion
					- `variant`: The variant of the algorithm to use. Defaults to SHA-512/256.
					- `encoding`: How to encode the hash as text. Can be base16 or base64.
						Defaults to base64.
				- `sha3`: Hash the redacted text with SHA-3 as with [`sha3`](\(urls.sha3)). Supports two optional parameters:
					- `variant`: The variant of the algorithm to use. Defaults to SHA3-512.
					- `encoding`: How to encode the hash as text. Can be base16 or base64.
						Defaults to base64.
```

**Review by jszwedko — APPROVED** [review]

File: website/cue/reference/remap/functions/redact.cue [hunk 3]
```diff
@@ -78,5 +111,33 @@ remap: functions: redact: {
...
+		},
+		{
+			title: "Replace with SHA-3 Hash",
+			source: #"""
+				redact("my id is 123456", filters: [r'\d+'], redactor: "sha3")
+				"""#
+			return: "my id is ZNCdmTDI7PeeUTFnpYjLdUObdizo+bIupZdl8yqnTKGdLx6X3JIqPUlUWUoFBikX+yTR+OcvLtAqWO11NPlNJw=="
+		},
+		{
+			title: "Replace with SHA-256 Hash using hex encoding",
```
Comment (maycmlee) [inline website/cue/reference/remap/functions/redact.cue:136]:
```suggestion
			title: "Replace with SHA-256 hash using hex encoding",
```

**Review by maycmlee — COMMENTED** [review]
Left a some suggestions.

**Review by pront — APPROVED** [review]
LGTM, depends on https://github.com/vectordotdev/vrl/pull/640

**Review by maycmlee — APPROVED** [review]

**Comment by pront** [comment]
> does this need anything for the changelog?

No, added the 'no-changelog' label.

**Pull request merged**